### PR TITLE
[feat](Nereids) Enhance filter processing in foreign key context to exclude hidden conjuncts

### PR DESCRIPTION
**Pull request by keanji-x**
## Proposed changes

Implement a new method isHiddenConjunct to identify hidden conjuncts based on delete sign and version columns. Update addFilter method to skip these hidden conjuncts when adding predicates to slotWithPredicates.

### CODE REVIEW

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/ForeignKeyContext.java [hunk 1]
```diff
@@ -136,13 +136,32 @@ void putAlias(Slot newSlot, Slot originSlot) {
         }
     }
 
+    private boolean isHiddenConjunct(Expression expression) {
+        for (Slot slot : expression.getInputSlots()) {
+            if (slot instanceof SlotReference) {
+                SlotReference slotReference = (SlotReference) slot;
+                if (slotReference.getColumn().isPresent()) {
+                    Column column = slotReference.getColumn().get();
+                    if (column.isDeleteSignColumn() || column.isVersionColumn()) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/ForeignKeyContext.java:145]: so version column still need?

**Comment by XieJiann** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/ForeignKeyContext.java (hunk 1 above)
Comment (XieJiann) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/ForeignKeyContext.java]: has removed

**Review by morrySnow — APPROVED** [review]

**Review by xzj7019 — APPROVED** [review]

**Pull request merged**